### PR TITLE
[shardformer] t5 support flash attention.

### PR DESCRIPTION
**Pull request by flybird1111**
## 📌 Checklist before creating the PR

- [x] I have created an issue for this PR for traceability
- [x] The title follows the standard format: `[doc/gemini/tensor/...]: A concise description`
- [x] I have added relevant tags if possible for us to better distinguish different PRs

## 🚨 Issue number

> Link this PR to your issue with words like fixed to automatically close the linked issue upon merge
>
> e.g. `fixed #1234`, `closed #1234`, `resolved #1234`

#4184 

## 📝 What does this PR do?

> Summarize your work here.
> if you have any plots/diagrams/screenshots/tables, please attach them here.

t5 support flash attention.

## 💥 Checklist before requesting a review

- [x] I have linked my PR to an issue ([instruction](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue))
- [x] My issue clearly describes the problem/feature/proposal, with diagrams/charts/table/code if possible
- [x] I have performed a self-review of my code
- [x] I have added thorough tests.
- [x] I have added docstrings for all the functions/methods I implemented

## ⭐️ Do you enjoy contributing to Colossal-AI?

- [ ] 🌝 Yes, I do.
- [ ] 🌚 No, I don't.

Tell us more if you don't enjoy contributing to Colossal-AI.

### CODE REVIEW

File: colossalai/shardformer/modeling/t5.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
+from typing import Optional, Tuple
+
+import torch
+
+__all__ = ['get_t5_forward']
+def get_t5_forward():
```
Comment (FrankLeeeee) [inline colossalai/shardformer/modeling/t5.py:6]: Leave a blank line.

File: colossalai/engine/schedule/_pipeline_schedule.py [hunk 2]
```diff
@@ -830,4 +830,4 @@ def _backward_step_helper(microbatch_id):
             output, label = pack_return_tensors(return_tensors)
             return output, label, accum_loss
         else:
-            return None, None, accum_loss
+            return None, None, accum_loss
```
Comment (FrankLeeeee) [inline colossalai/engine/schedule/_pipeline_schedule.py:833]: Use precommit please.

Comment (flybird1111) [reply colossalai/engine/schedule/_pipeline_schedule.py:833]: I have used pre-commit. but it still appeared.

Comment (flybird1111) [reply colossalai/engine/schedule/_pipeline_schedule.py]: > Use precommit please.

Updated

**Review by FrankLeeeee — APPROVED** [review]

**Pull request merged**